### PR TITLE
Remove redundant TODOs

### PR DESCRIPTION
**Pull request by SataQiu**
**What this PR does / why we need it**:
Add APIServerEndpoint validation for JoinConfiguration

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: controllers/kubeadmconfig_controller.go [hunk 1]
```diff
@@ -398,8 +398,8 @@ func (r *KubeadmConfigReconciler) reconcileDiscovery(cluster *clusterv1.Cluster,
 	}
 
 	// if BootstrapToken already contains an APIServerEndpoint, respect it; otherwise inject the APIServerEndpoint endpoint defined in cluster status
-	//TODO(fp) might be we want to validate user provided APIServerEndpoint and warn/error if it doesn't match the api endpoint defined at cluster level
 	apiServerEndpoint := config.Spec.JoinConfiguration.Discovery.BootstrapToken.APIServerEndpoint
+	needValidateAPIEndpoints := true
```
Comment (chuckha) [inline controllers/kubeadmconfig_controller.go]: We could do away with this boolean and always validate, right? I think the optimization of running this validation code is not worth introducing the boolean and we should validate every time regardless.

File: controllers/kubeadmconfig_controller.go [hunk 2]
```diff
@@ -409,6 +409,21 @@ func (r *KubeadmConfigReconciler) reconcileDiscovery(cluster *clusterv1.Cluster,
 		apiServerEndpoint = fmt.Sprintf("%s:%d", cluster.Status.APIEndpoints[0].Host, cluster.Status.APIEndpoints[0].Port)
 		config.Spec.JoinConfiguration.Discovery.BootstrapToken.APIServerEndpoint = apiServerEndpoint
 		log.Info("Altering JoinConfiguration.Discovery.BootstrapToken", "APIServerEndpoint", apiServerEndpoint)
+		needValidateAPIEndpoints = false
+	}
+	// validate user provided APIServerEndpoint matches the api endpoint defined at cluster level
+	if needValidateAPIEndpoints {
+		found := false
```
Comment (chuckha) [inline controllers/kubeadmconfig_controller.go]: I'd put this whole thing in a function,

```
func validateAPIEndpoints(endpoint string, endpoints []string) error {
...
}
```

**Comment by SataQiu** [comment]
Thank you for your explanation @fabriziopandini 
I understand your point. Let's ignore it for the moment.

**Comment by chuckha** [comment]
That's fine with me. Approving and lgtm as this is only comment changes.

/approve
/lgtm

**Pull request merged**